### ISSUE TITLE
Extend `create_csv` to use as cache

### ISSUE THREAD

**Issue by eliocamp** [boshek/rsoi#18]
It would be useful if all the `download_x()` functions to cache the results. It would be polite with the various servers (so that there would be no need to re-download constantly in exploratory analysis) and if made persistent, it would make the code reproducible and usable in situations with no internet. 

My proposal would be to memoise all functions in memory and also add an option to store the result in a file. Of course, a `cache = FALSE` option would force to download the fresh data.

**Comment by boshek** [comment]
memoise ✔️ 

caching: any thoughts on how to do this? I've [storr](https://github.com/richfitz/storr) in the past

**Comment by eliocamp** [comment]
I think caching to file could be done with a simple .csv file with name supplied by the user. Is simple and it allows user to then use the data as they see fit.

**Comment by boshek** [comment]
Ahh I see. CRAN policies typically require a package to get permission of the user before they write to disk. My interpretation of this here is that setting `create_csv` to TRUE is granting that permission. Is the proposal here to change the `create_csv` argument to just `cache`? I'd be open to that but I think it might be best to have the default still be FALSE. Apologies if I am misunderstanding here.

**Comment by eliocamp** [comment]
I don't know if I would change the `create_csv` argument; just to respect backwards compatibility. My idea would be to have two options. One that turns cache on or off and the other that tells 

``` r
download_soi(cache = FALSE, file = NULL)  # default. Downloads data every time
download_soi(cache = TRUE,  file = NULL)  # caches to memory. Downloads data only once per session.
download_soi(cache = FALSE, file = "data/soi.csv")  # downloads to file but still downloads data every time.
download_soi(cache = TRUE,  file = "data/soi.csv")  # downloads to file only if file doesn't exist.
```

The first two would be useful for simple analysis for which you don't necessarily want to create new files and have all the fuss about managing data. The other two be useful for code portability or for script to just download data locally (I have some of them to fetch reanalysis data, for example).

**Comment by boshek** [comment]
Yes. I am with you now. What do you think about a `.rsoi-data` folder rather than a `data` folder? Because a `data` folder is so common I just wonder about the etiquette of placing something in there for someone.

**Comment by eliocamp** [comment]
I was thinking that the location of the file would be decided by the user. So in my example, it is the user who decided that the csv file would be in data/soi.csv.

**Comment by boshek** [comment]
Closed via #22
